src/extensions/default/BrambleUrlCodeHints/main.js

```diff
@@ -155,13 +155,48 @@ define(function (require, exports, module) {
                         return;
                     }
 
+                    function getFilesRecursively(targetDir, callback) {
+                    var directory = FileSystem.getDirectoryForPath(targetDir);
```

Comment:
Before I can review this code, you need to fix all the indenting issues.  Throughout the code you've written, you've neglected to indent lots of blocks.  This is something you need to pay attention to, as it makes code more readable, and easier to understand.